base/dict.jl

```diff
@@ -325,8 +317,11 @@ function ht_keyindex2!(h::Dict{K,V}, key) where V where K
                 # in case "key" already exists in a later collided slot.
                 avail = -index
             end
-        elseif key === keys[index] || isequal(key, keys[index])
-            return index
+        else
+            p = pairs[index]
+            if key === p.first || isequal(key, p.first)
```

Comment:
To be consistent with above:
```suggestion
            k = pairs[index].first
            if key === k || isequal(key, k)
```